db/schema_tables.cc

```diff
@@ -1275,6 +1280,13 @@ static future<> do_merge_schema(distributed<service::storage_proxy>& proxy, std:
     auto new_aggregates = co_await read_schema_for_keyspaces(proxy, AGGREGATES, keyspaces);
     auto new_scylla_aggregates = co_await read_schema_for_keyspaces(proxy, SCYLLA_AGGREGATES, keyspaces);
 
+    if (has_tablet_mutations) {
+        slogger.info("Tablet metadata changed");
+        co_await proxy.local().get_db().invoke_on_all([&] (replica::database& db) -> future<> {
+            co_await db.get_notifier().update_tablet_metadata();
+        });
+    }
```

Comment:
What if we have a failure later? Doesn't this have to be transactional?

Reply:
It is transactional by the means of the group0 raft state machine. This is part of raft command (or snapshot) application, and raft will not let another read_barrier() to succeed locally before this succeeds.  

Currently, if this fails the raft implementation stops and you need a restart to trigger log replay.

Reply:
This sounds bad. A bad_alloc can kill the node? We need to add retries and an automatic restart.

In any case, I believe it's in the scope of raft, not this.